catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/ListApplication.java

```diff
@@ -0,0 +1,369 @@
+      Consumer<String> idConsumer,
+      Consumer<String> errorMessageConsumer,
+      StorableResource storableResource) {
+
+    try /*(StorableResource sr = storableResource)*/ {
+      store(getCreateInfo(storableResource), idConsumer, errorMessageConsumer);
+    } catch (IOException e) {
+      LOGGER.debug("Unable to create CreateInfo: ", e);
+    }
+  }
+
+  private Stream<StorableResource> createStream(CreateInfo createInfo, Splitter splitter)
+      throws IOException {
+    return splitter.split(createStorableResource(createInfo));
+  }
+
+  private StorableResource createStorableResource(CreateInfo createInfo) throws IOException {
+    return new StorableResource(createInfo.stream, createInfo.contentType, createInfo.filename);
+  }
+
+  private void appendMessageIfError(List<String> errorMessages, StorableResource storableResource) {
+    if (storableResource.isError()) {
+      errorMessages.add(storableResource.getErrorMessage());
+    }
+  }
+
+  private CreateInfo getCreateInfo(StorableResource storableResource) throws IOException {
+    CreateInfo createInfoItem = new CreateInfo();
+
+    createInfoItem.setFilename(storableResource.getFilename());
+    createInfoItem.setStream(storableResource.getInputStream());
+    createInfoItem.setContentType(
+        storableResource
+            .getMimeType()
+            .orElse(contentTypeFromFilename(storableResource.getFilename())));
+
+    return createInfoItem;
+  }
+
+  private CreateInfo parseAttachments(List<Part> contentParts) {
+
+    if (contentParts.size() == 1) {
+      Part contentPart = contentParts.get(0);
+      return parseAttachment(contentPart);
+    }
+    return null;
+  }
+
+  private CreateInfo parseAttachment(Part contentPart) {
+    CreateInfo createInfo = new CreateInfo();
+    InputStream stream;
+    String filename;
+    String contentType = null;
+
+    try {
+      stream = contentPart.getInputStream();
+      if (stream != null && stream.available() == 0) {
+        stream.reset();
+      }
+      createInfo.setStream(stream);
+    } catch (IOException e) {
+      LOGGER.info("IOException reading stream from file attachment in multipart body", e);
+    }
+
+    if (contentPart.getContentType() != null) {
+      contentType = contentPart.getContentType();
+    }
+
+    filename = contentPart.getSubmittedFileName();
+
+    if (StringUtils.isEmpty(filename)) {
+      LOGGER.debug("No filename parameter provided - generating default filename");
+      String fileExtension = DEFAULT_FILE_EXTENSION;
+      try {
+        fileExtension = mimeTypeMapper.getFileExtensionForMimeType(contentType); // DDF-2307
+        if (StringUtils.isEmpty(fileExtension)) {
+          fileExtension = DEFAULT_FILE_EXTENSION;
+        }
+      } catch (MimeTypeResolutionException e) {
+        LOGGER.debug("Exception getting file extension for contentType = {}", contentType);
+      }
+      filename = DEFAULT_FILE_NAME + "." + fileExtension; // DDF-2263
```

Comment:
Should this be a "todo" comment

Reply:
Sure. BTW, this came over from RESTEndpoint, but I'll change that class as well.